src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -107,9 +109,12 @@ public AsyncJsonTokenParser(
   private ListenableFuture<List<Verifier>> provideVerifiers(JsonToken jsonToken) {
     ListenableFuture<List<Verifier>> futureVerifiers;
     try {
-      futureVerifiers = asyncVerifierProviders
-          .getVerifierProvider(jsonToken.getSignatureAlgorithm())
-          .findVerifier(jsonToken.getIssuer(), jsonToken.getKeyId());
+      SignatureAlgorithm signatureAlgorithm = jsonToken.getSignatureAlgorithm();
+      AsyncVerifierProvider provider = asyncVerifierProviders.getVerifierProvider(signatureAlgorithm);
+      if (provider == null) {
+        throw new IllegalStateException("No valid provider for the algorithm: " + signatureAlgorithm);
```

Comment:
Sorry, I just noticed we use IllegalArgumentException for unsupported signature algorithm (when the alg value in the JWT is not recognized in the SignatureAlgorithm enum), so let's use IllegalArgumentException here as well (when the alg value in the JWT is recognized in the enum but the server doesn't support it because it doesn't provide a verifier for it).
```suggestion
        throw new IllegalArgumentException("Signature algorithm not supported: " + signatureAlgorithm);
```